brightray/browser/browser_main_parts.cc

```diff
@@ -252,11 +245,35 @@ void BrowserMainParts::PreMainMessageLoopStart() {
 }
 
 void BrowserMainParts::PreMainMessageLoopRun() {
+  auto* command_line = base::CommandLine::ForCurrentProcess();
+
+  // TODO(poiru): Link base_switches.cc and use
+  // switches::k{Enable,Disable}Features instead.
+  const auto enable_features =
+      command_line->GetSwitchValueASCII("enable-features");
+  auto disable_features = command_line->GetSwitchValueASCII("disable-features");
+
+  // TODO(deepak1556): Disable guest webcontents based on OOPIF feature.
+  // Disable surface synchronization and async wheel events to make OSR work.
+  disable_features +=
+      ",GuestViewCrossProcessFrames,SurfaceSynchronization,AsyncWheelEvents";
+
+  std::unique_ptr<base::FeatureList> feature_list(new base::FeatureList);
+  feature_list->InitializeFromCommandLine(enable_features, disable_features);
+
+  // FeatureList will have already been initialized by this point. We need
+  // to clear the existing instance before we can set a new one. Ideally we
+  // would initialize the first FeatureList instance, but cannot do this any
+  // earlier because we want to give the user JS script a chance to alter
+  // command-line switches.
+  base::FeatureList::ClearInstanceForTesting();
```

Comment:
We can override `BrowserMainParts::ShouldContentCreateFeatureList` to signal that we handle feature list creation. Its better to avoid using internal testing apis.

Reply:
Done, but we still need to use `ClearInstanceForTesting()` unfortunately. See updated comments.